src/applications/gi/reducers/calculator.js

```diff
@@ -47,14 +47,57 @@ export default function (state = INITIAL_STATE, action) {
       ].includes(field);
 
       if (isDollarAmount && !isFinite(value)) {
-      // Strip all non-numeric characters.
+        // Strip all non-numeric characters.
         convertedValue = +value.replace(/[^0-9.]+/g, '');
       }
 
-      const newState = {
+      let newState = {
         [field]: convertedValue,
       };
 
+      if (field === 'yellowRibbonDegreeLevel') {
+        const {
+          yellowRibbonPrograms
+        } = state;
+
+        const yellowRibbonDivisionOptions = yellowRibbonPrograms.length > 0 ?
+          [...new Set(yellowRibbonPrograms
+            .filter(program => program.degreeLevel === value)
+            .map(program => program.divisionProfessionalSchool))] :
+          [];
+
+        const yellowRibbonAmount = yellowRibbonPrograms
+          .find(program =>
+            program.degreeLevel === value &&
+            program.divisionProfessionalSchool === yellowRibbonDivisionOptions[0])
+          .contributionAmount;
+
+        newState = {
+          ...newState,
+          yellowRibbonDivisionOptions,
+          yellowRibbonDivision: { value: yellowRibbonDivisionOptions, label: yellowRibbonDivisionOptions },
+          yellowRibbonAmount
+        };
+      }
+
+      if (field === 'yellowRibbonDivision') {
```

Comment:
When the division is updated, this updates

- the value in the yellow ribbon contribution amount